schemas/EGA.analysis.json

```diff
@@ -5,9 +5,190 @@
+              "enum": [ "sequence variation", "sequence alignment", "phenotype characterization", "processed sequencing data", "processed array data", "sequence annotation", "sequence assembly" ],
+              "meta:enum": [ "sequence variation__ERO:0100211 - Analysis of variations at specific loci in the genomes of organisms (mutation and polymorphism) across or within a species, population, or individual (e.g healthy vs diseased tissue).", "sequence alignment__ERO:0100032 - objective to display graphically how the sequences of two or more macromolecules align along a linear axis.", "phenotype characterization__ERO:0000923 - The result of an organismal assay that involves characterization of a phenotype; any observable characteristic or trait of an organism: such as its morphology, development, biochemical or physiological properties, behavior, and products of behavior (such as a bird's nest). Phenotypes result from the expression of an organism's genes as well as the influence of environmental factors and the interactions between the two.", "processed sequencing data__ - Analysis in which raw sequencing data (e.g. FastQ files) were processed in any way (e.g. normalization, noise reduction, alignment...) and transformed into processed genotype data files [EFO:0004663].", "processed array data__ - Analysis in which raw array data (e.g. CEL files) were processed in any way (e.g. normalization, noise reduction...) and transformed into processed array data files [EFO:0004096].", "sequence annotation__ - For submitting sequence annotation files, which are usually 'tab' files. Examples include gene count and OTU tables from metagenomic studies.", "sequence assembly__topic:0196 - The assembly of fragments of a DNA sequence to reconstruct the original sequence." ]            }
+          },
+          "reference_alignment_details": {
+            "type": "array",
+            "title": "Reference assembly and sequence details",
+            "description": "Node containing details of the reference sequence used in the alignment of raw sequences.",
+            "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/reference_alignment_details"
+          }
+        },
+        "allOf": [
+          {
+            "title": "If the analysis type includes a sequence comparison, the reference is expected",
+            "if": {
+              "required": ["analysis_types"],
+              "properties": {
+                "analysis_types": {
+                  "type": "array",
+                  "minItems": 1,
+                  "contains": {
+                    "type": "string",
+                    "enum": [ "sequence variation", "sequence alignment"]
```

Comment:
I'm not sure what type of analysis this block is referencing. If it means any type of analysis where an alignment step is done, one could also be doing a sequence annotation or a sequence assembly analysis.

Reply:
But since it's a constraint placed on those that **always** have a reference, I only enumerated Seq. variation and alignment. Correct me if I'm wrong, but can't assemblies and annotations be _de novo_ (without a reference)?

Reply:
- Assemblies --> can be done by aligning experiments reads to a reference OR aligning experiment reads to themselves
- Annotations --> can be done with or without a reference
- Variation --> can be done with a reference (what variation is in my samples compared to a known reference) OR without a reference (what variation exists among all my sequenced samples)
- Alignment --> can be done with a reference (e.g. HISAT) OR without a reference (e.g. clustering/aligning reads to each other, not a reference, to quantify )

So really, none of these analysis approaches _requires_ a reference. They can all be done without one. It may be easy in this first round to remove this constraint.

Reply:
Well the constraint only applies for ``sequence variation`` and ``sequence alignment`` but it's true that there are cases in which the external reference of an assembly is not needed. 

But even if the assembly reference is not always used, the sequence against which some other sequences are compared is always there. Should we modify the ``reference_alignment_details`` node to have some kind of explanation (e.g. what sequence is used as ref. if not an assembly), or expect that in the protocols for an alignment data transformation?

Reply:
I suggest making this filed more conservative / less restrictive. Provide an option for the user to supply a reference, if they had one, but don't but constraints on the assumptions because science is messy :)